conda/core/index.py

```diff
@@ -47,39 +48,57 @@
 fail_unknown_host = False
 
 
-def supplement_index_with_prefix(index, prefix, channel_priority_map):
-    # type: (Dict[Dist, IndexRecord], str, Dict[channel_url, Tuple[canonical_name, priority]) -> None  # NOQA
+def supplement_index_with_prefix(index, prefix, channels):
+    # type: (Dict[Dist, IndexRecord], str, Set[canonical_channel]) -> None  # NOQA
     # supplement index with information from prefix/conda-meta
     assert prefix
-
-    priorities = {chnl: prrty for chnl, prrty in itervalues(channel_priority_map)}
-    maxp = max(itervalues(priorities)) + 1 if priorities else 1
+    maxp = len(channels) + 1
     for dist, info in iteritems(linked_data(prefix)):
-        fn = info['fn']
```

Comment:
This old dist string construction code should be removed, right? I mean, dist is available right there in `linked_data`.